### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by lapp0**
Rendered: https://lapp0.github.io/outlines/quickstart/

# Changes
- Create `model.md` documentation with feature matrix
- Update `quickstart.md` so its more concise, and links to relevant `reference/generation/index.md` and `reference/models/index.md` documentation pages. Moves most structured generation examples to `reference/generation/index.md`
- Include `structured_generation_explanation.md` which details automata construction and logits processing.
- Use `phi-3-mini` instead of `mistral` in all pages except external libraries cookbooks because it doesn't require an access token and is a more performant model for its size.
- Add Remi's AI Engineer World’s Fair Presentation

Fixes 
- Fixes https://github.com/outlines-dev/outlines/issues/675
- Fixes https://github.com/outlines-dev/outlines/issues/477
- Fixes https://github.com/outlines-dev/outlines/issues/1007

### CODE REVIEW

**Comment by rlouf** [comment]
Thanks for the doc update! Two comments:

- We can remove Mistral-7B from the README as well. I also saw it in a couple of places in the rendered doc.
- For the talk we'll have to do with the current video for now.

**Comment by lapp0** [comment]
Updated docs
- added Remi's AI Engineer World’s Fair Presentation
- removed `mistralai/Mistral-7B-*` everywhere
- kept documentation which uses quants, e.g. `TheBloke/Mistral-7B-Instruct-v0.2-GPTQ`
- kept external library guides in cookbook using mistral, e.g.
```
docs/cookbook/simtom.md:MODEL_NAME = "mistral-community/Mistral-7B-Instruct-v0.3"
```

**Comment by rlouf** [comment]
We need to fix the merge conflict and then it should be ready to merge.
